program/clientcommon/runtime.go

```diff
@@ -65,6 +65,14 @@ func InitRuntime(binpath, sockpath, dbpath string) *eval.Evaler {
 			version, err := cl.Version()
 			if err != nil {
 				fmt.Fprintln(os.Stderr, "warning: socket exists but not responding version RPC:", err)
+				fmt.Fprintln(os.Stderr, "warning: removing ", sockpath, " and restarting daemon !")
+				err3 := os.Remove(sockpath) // If the sock file exists , but doesn't have a daemon attached to it , remove it .
+				if err3 != nil {
+					fmt.Fprintln(os.Stderr, "warning: deletion of", sockpath, " failed! \n warning: remove it manually for the daemon to work correctly !", err3)
+				}
+				fmt.Fprintln(os.Stderr, "[!]attempting to restart daemon .")
+				InitRuntime(toSpawn.BinPath, toSpawn.SockPath, toSpawn.DbPath)
```

Comment:
Calling `InitRuntime` itself is no better than a `goto` to the beginning of the function, please do not do that.

A way to simplify the control flow in the function would be to factor out two functions `detectDaemon` and `spawnDaemon`, and change the code here to:

```go
if !detectDaemon() {
  spawnDaemon()
}
```